File: app/src/main/java/com/example/babajidemustapha/survey/features/newsurvey/fragments/CreateSurveyFragment.java
package com.example.babajidemustapha.survey.features.newsurvey.fragments;


import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Switch;
import android.widget.Toast;

import androidx.fragment.app.Fragment;

import com.example.babajidemustapha.survey.R;
import com.example.babajidemustapha.survey.features.newsurvey.activities.CreateQuestions;


/**
 * A simple {@link Fragment} subclass.
 * Use the {@link CreateSurveyFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class CreateSurveyFragment extends Fragment {
    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param1";
    private static final String ARG_PARAM2 = "param2";

    // TODO: Rename and change types of parameters
    private String mParam1;
    private String mParam2;
    EditText name;
    EditText desc;
    Switch privacy;
    Button btn;

    public CreateSurveyFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param1 Parameter 1.
     * @param param2 Parameter 2.
     * @return A new instance of fragment CreateSurveyFragment.
     */
    // TODO: Rename and change types and number of parameters
    public static CreateSurveyFragment newInstance(String param1, String param2) {
        CreateSurveyFragment fragment = new CreateSurveyFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param1);
        args.putString(ARG_PARAM2, param2);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam1 = getArguments().getString(ARG_PARAM1);
            mParam2 = getArguments().getString(ARG_PARAM2);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View layout = inflater.inflate(R.layout.fragment_create_survey, container, false);
        getActivity().setTitle("Survey Configuration");
        name = layout.findViewById(R.id.name);
        desc = layout.findViewById(R.id.desc);
        privacy = layout.findViewById(R.id.privacy);
        btn = layout.findViewById(R.id.go);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(!name.getText().toString().isEmpty() || !desc.getText().toString().isEmpty()){
                    Intent intent = new Intent(getActivity(), CreateQuestions.class);
                    intent.putExtra("survey name",name.getText().toString());
                    intent.putExtra("survey desc",desc.getText().toString());
                    intent.putExtra("survey privacy",privacy.isChecked());
                    startActivity(intent);
                }
                else{
                    Toast.makeText(getContext(),"1 or more fields empty",Toast.LENGTH_SHORT).show();
                }
            }
        });
        return layout;
    }


}

File: app/src/main/java/com/example/babajidemustapha/survey/features/newsurvey/fragments/CreateSurveyFragment.kt
package com.example.babajidemustapha.survey.features.newsurvey.fragments


import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.example.babajidemustapha.survey.R
import com.example.babajidemustapha.survey.features.newsurvey.fragments.QuestionsSetupFragment.OnSurveySetupInteractionListener
import com.example.babajidemustapha.survey.shared.room.entities.Survey
import kotlinx.android.synthetic.main.fragment_create_survey.*


/**
 * A simple [Fragment] subclass.
 * Use the [CreateSurveyFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class CreateSurveyFragment : Fragment() {

    private var listener: OnSurveySetupInteractionListener? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_create_survey, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        activity?.title = "Create Survey"
        go.setOnClickListener { v ->

            if (!name.text.toString().isEmpty() && !desc.text.toString().isEmpty()) {
                val survey = Survey()
                survey.name = name.text.toString()
                survey.desc = desc.text.toString()
                survey.isPrivacy = privacy.isChecked
                listener?.createSurvey(survey)
            } else {
                Toast.makeText(context, "1 or more fields empty", Toast.LENGTH_SHORT).show()
            }
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is OnSurveySetupInteractionListener) {
            listener = context
        } else {
            throw RuntimeException("$context must implement OnSurveySetupInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        listener = null
    }

    companion object {
        fun newInstance(): CreateSurveyFragment = CreateSurveyFragment()
    }


}// Required empty public constructor
